src/helpers/retry.ts

```diff
@@ -0,0 +1,28 @@
+import retry from 'async-retry'
+
+const errorCodesNoRetry = [400, 403, 413]
+
+export const retryRequest = async (requestBuilder: () => Promise<any>, retryOpts: retry.Options): Promise<void> => {
+  // Request function, passed to async-retry
+  const doRequest = async (bail: (e: Error) => void) => {
+    try {
+      await requestBuilder()
+    } catch (error) {
+      if (error.response) {
+        // If it's an axios error
+        if (!errorCodesNoRetry.includes(error.response.status)) {
+          // And a status code that is not excluded from retries, throw the error so that upload is retried
+          throw error
+        }
+      } else {
+        // If it's another error or an axios error let us retry just in case
```

Comment:
For the sake of clarity what do you think about the following:

```typescript
if (error.response) {
    if (errorCodesNoRetry.includes(error.response.status)) {
        bail(error)
    }
}
// Other cases are retried: other axios errors as well as
// non-axios errors such as DNS resolution errors and connection timeouts
throw error

```

Reply:
Fixed